core/chain_manager.go

```diff
@@ -84,13 +90,24 @@ type ChainManager struct {
 }
 
 func NewChainManager(chainDb common.Database, pow pow.PoW, mux *event.TypeMux) (*ChainManager, error) {
-	cache, _ := lru.New(blockCacheLimit)
+	headerCache, _ := lru.New(headerCacheLimit)
```

Comment:
Debate it with @obscuren, he's the one who originally added the cache :P. I just extended it for the split database layout.